### ISSUE TITLE
Pact magic slots forced on warlock multiclass, shift click not working to cast with other slots

### ISSUE THREAD

**Issue by hugh-jaccident** [SolastaMods/SolastaUnfinishedBusiness#5069]
Solasta version: 1.5.97
UB version: 1.5.97.39

Bug description: I have a celestial warlock with some levels in sorcerer, I'd like to be able to use a lvl 1 sorcerer slot to cast guiding bolt from my warlock list, however it always uses a pact magic slot, even when I shift click as it says to do in the settings. Furthermore, when I get the reaction prompt for hellish rebuke, it shows the option to use a sorcerer slot, however, when I select this it still ends up using a pact magic slot in the end. 

To reproduce: Attempt to cast a warlock spell using a sorcerer slot, either through the reaction prompt or by shift clicking a warlock spell

Files attached in zip, log folder was empty
[pact magic bug files.zip](https://github.com/user-attachments/files/17844006/pact.magic.bug.files.zip)

**Comment by hugh-jaccident** [comment]
Shield is a spell from the character's sorcerer list though

**Comment by ThyWoof** [comment]
Yep, and if consumed a pact slot. Working as if should. I tested all other scenarios. This is something that hasn’t changed for a couple of years now.

**Comment by ThyWoof** [comment]
I’m not in the QA business. If you would like me to test your scenario get me a save on the hero. My patience is thin to wait an entire battle turn to test non existing issues.

**Comment by hugh-jaccident** [comment]
I'm sorry to test your patience but I don't know what to tell you, it isn't
working on my end. Specifically the warlock spells aren't working with the
sorcerer slots. I'll send you another save later. Again, I'm not trying to
waste your time, I assure you the issue I'm having is real.

**Comment by ThyWoof** [comment]
I trust you but something I have no control at all. If I had to guess, some windows setting causing this. Stick shift key, etc. Mod is coded to intercept any SHIFT pressed, left or right.

**Comment by hugh-jaccident** [comment]
[new solasta save.zip](https://github.com/user-attachments/files/17854096/new.solasta.save.zip)
This is a save where Natalie is about to be attacked, when I try to cast hellish rebuke at 1st level, it rolls the damage as if it's first level, but consumes a pact magic slot. If the issue doesn't happen on your end then I guess I'm just cursed. I think you might be right about some windows setting messing with my shift key though.

**Comment by hugh-jaccident** [comment]
So here's a save where you shouldn't  have to go through a round of combat. Natalie is about to be attacked by the warlord, if you try to cast hellish rebuke using a level 1 slot, you can check on her following turn that a pact magic slot has been consumed instead. You're correct that I can cast shield using the pact magic slot just fine, but the reverse isn't true for me. 
[new solasta save.zip](https://github.com/user-attachments/files/17854581/new.solasta.save.zip)
